pkg/compactv2/modifiers.go

```diff
@@ -328,4 +333,105 @@ func (p *delChunkSeriesIterator) Next() bool {
 
 func (p *delChunkSeriesIterator) At() chunks.Meta { return p.curr }
 
-// TODO(bwplotka): Add relabelling.
+type RelabelModifier struct {
+	relabels []*relabel.Config
+}
+
+func WithRelabelModifier(relabels ...*relabel.Config) *RelabelModifier {
+	return &RelabelModifier{relabels: relabels}
+}
+
+func (d *RelabelModifier) Modify(_ index.StringIter, set storage.ChunkSeriesSet, log ChangeLogger, p ProgressLogger) (index.StringIter, storage.ChunkSeriesSet) {
+	// Gather symbols.
+	symbols := make(map[string]struct{})
+	// TODO(yeya24): use a map to store the label set and the chunk iterators
+	// since multiple series might be relabelled to the same series.
+	seriesSets := make([]storage.ChunkSeries, 0)
+
+SeriesLoop:
+	for set.Next() {
+		s := set.At()
+		lbls := s.Labels()
+
+		if processedLabels := relabel.Process(lbls, d.relabels...); len(processedLabels) == 0 {
+			// Special case: Delete whole series.
+			chksIter := s.Iterator()
+			var (
+				minT int64 = math.MaxInt64
+				maxT int64 = math.MinInt64
+			)
+			for chksIter.Next() {
+				c := chksIter.At()
+				if c.MinTime < minT {
+					minT = c.MinTime
+				}
+				if c.MaxTime > maxT {
+					maxT = c.MaxTime
+				}
+			}
+
+			if err := chksIter.Err(); err != nil {
+				return errorOnlyStringIter{err: err}, nil
+			}
+
+			var deleted tombstones.Intervals
+			// If minTime is set then there is at least one chunk.
+			if minT != math.MaxInt64 {
+				deleted = deleted.Add(tombstones.Interval{Mint: minT, Maxt: maxT})
+			}
+			log.DeleteSeries(lbls, deleted)
+			p.SeriesProcessed()
+			continue SeriesLoop
```

Comment:
We don't need to use `continue SeriesLoop` because the else part won't be executed.

Reply:
Oh, good catch!